File: example/Simplace_Soil_Temperature/src/java/Simplace_Soil_Temperature/SoilTemperatureRate.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoilTemperatureRate
{
    private double rSnowWaterContentRate;
    private double rSoilSurfaceTemperatureRate;
    private Integer rAgeOfSnowRate;
    
    public SoilTemperatureRate() { }
    
    public SoilTemperatureRate(SoilTemperatureRate toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.rSnowWaterContentRate = toCopy.getrSnowWaterContentRate();
            this.rSoilSurfaceTemperatureRate = toCopy.getrSoilSurfaceTemperatureRate();
            this.rAgeOfSnowRate = toCopy.getrAgeOfSnowRate();
        }
    }
    public double getrSnowWaterContentRate()
    { return rSnowWaterContentRate; }

    public void setrSnowWaterContentRate(double _rSnowWaterContentRate)
    { this.rSnowWaterContentRate= _rSnowWaterContentRate; } 
    
    public double getrSoilSurfaceTemperatureRate()
    { return rSoilSurfaceTemperatureRate; }

    public void setrSoilSurfaceTemperatureRate(double _rSoilSurfaceTemperatureRate)
    { this.rSoilSurfaceTemperatureRate= _rSoilSurfaceTemperatureRate; } 
    
    public Integer getrAgeOfSnowRate()
    { return rAgeOfSnowRate; }

    public void setrAgeOfSnowRate(Integer _rAgeOfSnowRate)
    { this.rAgeOfSnowRate= _rAgeOfSnowRate; } 
    
}

File: example/Simplace_Soil_Temperature/src/cpp/Simplace_Soil_Temperature/SoilTemperatureRate.cpp
#include "SoilTemperatureRate.h"
using namespace Simplace_Soil_Temperature;

SoilTemperatureRate::SoilTemperatureRate() {}

double SoilTemperatureRate::getrSnowWaterContentRate() { return this->rSnowWaterContentRate; }
double SoilTemperatureRate::getrSoilSurfaceTemperatureRate() { return this->rSoilSurfaceTemperatureRate; }
int SoilTemperatureRate::getrAgeOfSnowRate() { return this->rAgeOfSnowRate; }

void SoilTemperatureRate::setrSnowWaterContentRate(double _rSnowWaterContentRate) { this->rSnowWaterContentRate = _rSnowWaterContentRate; }
void SoilTemperatureRate::setrSoilSurfaceTemperatureRate(double _rSoilSurfaceTemperatureRate) { this->rSoilSurfaceTemperatureRate = _rSoilSurfaceTemperatureRate; }
void SoilTemperatureRate::setrAgeOfSnowRate(int _rAgeOfSnowRate) { this->rAgeOfSnowRate = _rAgeOfSnowRate; }